### PR TITLE
Apikit 2771 - Create configuration files per environment.

### PR DESCRIPTION
**Pull request by nicolaskosacoff**

### CODE REVIEW

File: src/main/java/org/mule/tools/apikit/model/Configuration.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.tools.apikit.model;
+
+import java.util.Map;
+
+public class Configuration {
+
+  private String environment;
+  private CommonProperties commonProperties;
```
Comment (dcourtil) [inline src/main/java/org/mule/tools/apikit/model/Configuration.java:14]: Which is the purpose or advantage of having a CommonProperties and HttpCommonProperty classes? Shouldn't be treated as regular properties. Maybe we can have a "Properties Group" concept to group them semantically, as the configurations.

File: src/main/java/org/mule/tools/apikit/model/CustomConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.tools.apikit.model;
+
+
+import java.util.Optional;
+
+public class CustomConfiguration {
+
+  private Optional<String> externalConfigurationFile;
```
[the remark this answers is not in the document]
Comment (nicolaskosacoff) [reply src/main/java/org/mule/tools/apikit/model/CustomConfiguration.java:14]: Deleted the whole file as it wasn't being used.

File: src/main/java/org/mule/tools/apikit/output/MuleConfigGenerator.java [hunk 3]
```diff
@@ -230,12 +240,121 @@ private MuleConfig getUpdatedMuleConfig(MuleConfig muleConfig, Element newApikit
...
+    MuleConfig global = createCommonPropertiesFile();
     for (ApikitMainFlowContainer api : apiContainers) {
-      muleConfigs.add(api.getMuleConfig() == null ? createMuleConfig(api) : api.getMuleConfig());
+      FlowScope flowScope = null;
+      String muleConfigID = createMuleConfigID(api.getId());
+      ConfigurationPropertiesConfig configurationPropertiesConfig = createConfigurationProperties();
+      APIAutodiscoveryConfig apiAutodiscoveryConfig =
+          createAPIAutodiscoveryConfig(FilenameUtils.removeExtension(muleConfigID).concat(MAIN_FLOW_SUFFIX));
+      MuleConfig muleConfig = createMuleConfig(api);
+      if (global != null) {
```
[the remark this answers is not in the document]
Comment (nicolaskosacoff) [reply src/main/java/org/mule/tools/apikit/output/MuleConfigGenerator.java:251]: The most I could extract is "addCustomizations" method that would work for all MuleConfig instances

File: src/main/java/org/mule/tools/apikit/model/Configuration.java (hunk 1 above)
Comment (nicolaskosacoff) [reply src/main/java/org/mule/tools/apikit/model/Configuration.java]: Refactor done.

**Comment by nicolaskosacoff** [comment]
Closing because this PR will be replaced by https://github.com/mulesoft/apikit-rest-scaffolder/pull/120

**Pull request closed without merging**